sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/LockScreenManager.java

```diff
@@ -0,0 +1,274 @@
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.util.Log;
+
+import com.smartdevicelink.api.BaseSubManager;
+import com.smartdevicelink.protocol.enums.FunctionID;
+import com.smartdevicelink.proxy.RPCNotification;
+import com.smartdevicelink.proxy.interfaces.ISdl;
+import com.smartdevicelink.proxy.rpc.OnDriverDistraction;
+import com.smartdevicelink.proxy.rpc.OnHMIStatus;
+import com.smartdevicelink.proxy.rpc.OnSystemRequest;
+import com.smartdevicelink.proxy.rpc.enums.DriverDistractionState;
+import com.smartdevicelink.proxy.rpc.enums.HMILevel;
+import com.smartdevicelink.proxy.rpc.enums.LockScreenStatus;
+import com.smartdevicelink.proxy.rpc.enums.RequestType;
+import com.smartdevicelink.proxy.rpc.listeners.OnRPCNotificationListener;
+import com.smartdevicelink.util.HttpUtils;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+
+/**
+ * <strong>LockscreenManager</strong> <br>
+ *
+ * Note: This class must be accessed through the SdlManager. Do not instantiate it by itself. <br>
+ *
+ * The LockscreenManager handles the logic of showing and hiding the lock screen. <br>
+ *
+ */
+public class LockScreenManager extends BaseSubManager {
+
+	private static final String TAG = "LockScreenManager";
+	private WeakReference<Context> context;
+	private HMILevel hmiLevel;
+	private boolean driverDistStatus, lockScreenEnabled, showOEMLogo;
+	private int lockScreenIcon, lockScreenColor, customView;
+	private OnRPCNotificationListener systemRequestListener, ddListener, hmiListener;
+	private String OEMIconUrl;
+	private Bitmap lockScreenOEMIcon;
+
+	public LockScreenManager(LockScreenConfig lockScreenConfig, Context context, ISdl internalInterface){
+
+		super(internalInterface);
+		this.context = new WeakReference<>(context);
+
+		// set initial class variables
+		hmiLevel = HMILevel.HMI_NONE;
+		driverDistStatus = false;
+
+		// setup the manager
+		readConfiguration(lockScreenConfig);
+		setupListeners();
+
+		// transition state
+		transitionToState(READY);
+	}
+
+	@Override
+	public void dispose(){
+		// send broadcast to close lock screen if open
+		context.get().sendBroadcast(new Intent(SDLLockScreenActivity.CLOSE_LOCK_SCREEN_ACTION));
+		// remove listeners
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_HMI_STATUS, hmiListener);
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_DRIVER_DISTRACTION, ddListener);
+		if (showOEMLogo) {
+			internalInterface.removeOnRPCNotificationListener(FunctionID.ON_SYSTEM_REQUEST, systemRequestListener);
+		}
+		lockScreenOEMIcon = null;
+		OEMIconUrl = null;
+
+		// transition state
+		transitionToState(SHUTDOWN);
+	}
+
+	////
+	// SETUP
+	////
+
+	/**
+	 * Reads configuration set in SDLManager Builder
+	 * @param lockScreenConfig - the configuration set.
+	 */
+	private void readConfiguration(LockScreenConfig lockScreenConfig){
```

Comment:
Does this method ever get called beyond the constructor? Might be best to just include this parsing in the constructor.